src/compiler.rs

```diff
@@ -75,14 +75,39 @@ impl Compiler {
               .join(Self::expand_tilde(&relative.cooked)?)
               .lexiclean();
 
-            if import.is_file() {
+            let import_base_str = import.to_string_lossy();
+            let has_glob = import_base_str.find('*').is_some();
+
+            if has_glob {
+              let glob_options = glob::MatchOptions {
+                case_sensitive: true,
+                require_literal_separator: false,
+                require_literal_leading_dot: false,
+              };
+              let import_paths = glob::glob_with(&import_base_str, glob_options)
+                .map_err(|error| Error::ImportGlob { error, path: *path })?;
+              for import in import_paths {
+                let Ok(import) = import else { continue };
+
+                if import.is_file() {
```

Comment:
I think a glob matching a directory should be an error. Just because we can backwards-compatibly make it not an error, but not backwards-compatibly turn it into an error.